.github/workflows/copilot_deploy.yml

```diff
@@ -43,74 +52,73 @@ jobs:
       owner: ${{ github.repository_owner }}
       application: funding-service-design-notification
 
-  pre_deploy_tests:
-    secrets:
-      E2E_PAT: ${{secrets.E2E_PAT}}
-      GOV_NOTIFY_API_KEY: ${{secrets.GOV_NOTIFY_API_KEY}}
-    uses: communitiesuk/funding-service-design-workflows/.github/workflows/pre-deploy.yml@main
-    with:
-      postgres_unit_testing: false
+  copilot_environments_workflow_setup:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.copilot_env_matrix.outputs.env_list }}
+    steps:
+      - id: copilot_env_matrix
+        run: |
+          if [ "${{ inputs.environment }}" != '' ]; then
+            echo "env_list=[\"${{ inputs.environment }}\"]" >> $GITHUB_OUTPUT
+          elif [ "${{ github.ref }}" == 'refs/heads/main' ]; then
+            echo "env_list=[\"dev\", \"test\", \"uat\", \"production\"]" >> $GITHUB_OUTPUT
+          else
+            echo "env_list=[\"dev\", \"test\"]" >> $GITHUB_OUTPUT
```

Comment:
Do we put it in test for every commit?

Reply:
It would have to - the post award tests should be run against the test environment, so the test code will need to be updated to run against it.

Post migration, we'll be creating an environment per ticket, but at the moment we just want an e2e experience